src/main/java/org/folio/rest/impl/CheckOutLockAPI.java

```diff
@@ -0,0 +1,187 @@
+import static org.folio.rest.persist.PostgresClient.convertToPsqlStandard;
+import static org.folio.support.DbUtil.rowSetToStream;
+
+public class CheckOutLockAPI implements CheckOutLockStorage {
+
+  private static final String CHECK_OUT_LOCK_TABLE = "check_out_lock";
+
+  private static final Logger log = LogManager.getLogger();
+
+  @Override
+  public void getCheckOutLockStorageByLockId(String lockId, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    log.debug("getCheckOutLockStorageByLockId:: getting lock with lockId {} ", lockId);
+    String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+    PostgresClient postgresClient = postgresClient(vertxContext, okapiHeaders);
+    if (!UuidUtil.isUuid(lockId)) {
+      asyncResultHandler.handle(new SucceededFuture<>(GetCheckOutLockStorageByLockIdResponse.respond400WithTextPlain("Invalid lock id")));
+      return;
+    }
+    postgresClient.execute(getLockByIdSql(lockId, tenantId), handler -> {
+      if (handler.succeeded()) {
+        asyncResultHandler.handle(succeededFuture(GetCheckOutLockStorageByLockIdResponse.respond200WithApplicationJson(this.mapToCheckOutLock(handler.result()))));
+      } else {
+        asyncResultHandler.handle(succeededFuture(GetCheckOutLockStorageByLockIdResponse.respond404()));
+      }
+    });
+  }
+
+  @Override
+  public void getCheckOutLockStorage(String userId, int offset, int limit,
+                                     Map<String, String> okapiHeaders,
+                                     Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    log.debug("getCheckOutLockStorage:: getting locks ");
+    String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+    PostgresClient postgresClient = postgresClient(vertxContext, okapiHeaders);
+    postgresClient.execute(getLocksSqlwithQueryParams(tenantId, userId, offset, limit), handler -> {
+      if (handler.succeeded()) {
+        asyncResultHandler.handle(succeededFuture(GetCheckOutLockStorageResponse.respond200WithApplicationJson(this.mapToCheckOutLocks(handler.result()))));
+      } else {
+        asyncResultHandler.handle(succeededFuture(GetCheckOutLockStorageResponse.respond422WithTextPlain("Invalid Parameters")));
+      }
+    });
+  }
+
+  @Override
+  public void postCheckOutLockStorage(CheckoutLockRequest entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    log.debug("postCheckOutLockStorage:: entity {} {} ", entity.getUserId(), entity.getTtlMs());
+    PostgresClient postgresClient = postgresClient(vertxContext, okapiHeaders);
+    String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+    postgresClient.execute(deleteOutdatedLockSql(entity.getUserId(), tenantId, entity.getTtlMs()), handler1 -> {
+      try {
+        if (handler1.succeeded()) {
```

Comment:
Should we give handler1 and handler2 more descriptive names? Like, deleteLockHandler, createLockHandler.

Reply:
handler name updated like mentioned above